### PR TITLE
Issue 372

### PR DESCRIPTION
**Pull request by mmoadeli**
initial work on addressing issue #372

[Review on Reviewable](https://reviewable.io/reviews/maidsafe/routing/383)

### CODE REVIEW

**Comment by maidsafe-highfive** [comment]
r? @inetic

(maidsafe_highfive has picked a reviewer for you, use r? to override)

**Comment by mmoadeli** [comment]
The work in no way is in review state

File: src/node_interface.rs [hunk 1]
```diff
@@ -116,7 +116,8 @@ pub trait Interface : Sync + Send {
                         data: Vec<u8>) -> Result<MessageAction, InterfaceError>;
 }
 
-pub trait CreatePersonas<F : Interface> : Sync + Send  {
-    fn create_personas(&mut self) -> F;
```
Comment (benjaminbollen) [inline src/node_interface.rs]: please start from master and don't pull in the commits from "simplifying routing_node creation"

**Comment by mmoadeli** [comment]
carrying over the relevant tests are done.  Writing new tests in not clear. More clarifications are needed

**Comment by mmoadeli** [comment]
in handle_refresh our_authority returns ManagedNode, and error returned as OurCloseGroup is expected.

**Pull request merged**